lib/generator_graph.ml

```diff
@@ -0,0 +1,149 @@
+let copy_matrix matrix =
+  Array.init (Array.length matrix) (fun i -> Array.copy matrix.(i))
+
+let check_neighbors_does_not_have_same_color grid height width current_row
```

Comment:
in this function, you don't check the modulo position ? ((0, 0 ) for (w - 1, 0) for exemple)